src/views/CoCurricularTranscript/index.tsx

```diff
@@ -36,12 +36,15 @@ const CoCurricularTranscript = () => {
     loadTranscript();
   }, [profile]);
 
-  if (loading || loadingProfile) {
```

Comment:
Why did you remove loading here? Is that not still a valid measure?

Reply:
So the problem is that if `transcriptService.getItems(...` fails, nothing catches the error and sets loading to false.

Instead of your updates to lines 44-47, I would add a `.catch((error) => {...})` to the call out to the service on line 31.

Reply:
This is an issue because the transcript request fails, so loading is never set to false. I would add a `.catch(...)`, which will set loading to false, to the transcript request on line 31 and then put loading back in here.

Reply:
I see. I add an else statement outside `if (loading || loadingProfile)`, and it solves the problem. Is this what you mean to do? However, I failed to use the `!useIsAuthenticated()` on line 40, it seems like it makes the `profile` on line 61 may be equal to null, which is not allowed. I won't have this problem if I use `!profile()`. Is there a reason that we should use `useIsAuthenticated` instead of `profile`?

![image](https://github.com/gordon-cs/gordon-360-ui/assets/78650688/35faf657-3ee3-4305-bbe1-8283e273ebb2)

Reply:
Using `!profile` as the first condition is not ideal because it will render the example profile for a signed-in user while their actual profile loads (which will either be confusing or just flashes of different content that doesn't look good).

You can't call `useIsAuthenticated()` like that because it's a Hook - you have to obey the Rules of Hooks (see the [orange Pitfall in the React Docs on hooks](https://react.dev/learn/state-a-components-memory#meet-your-first-hook)).

What you can do is something like the following:
```javascript
const isAuthenticated = useIsAuthenticated();
const {profile: userProfile, loading: loadingProfile) = useUser();
const profile = isAuthenticated ? userProfile : exampleProfile;
...
const loadTranscript = () => {
  if (isAuthenticated) {
    if (!profile) return;
    setLoading(true);
    // load transcript
  } else {
    setTranscript(exampleTranscript);
  }
  setLoading(false);
}
...
if (loading || loadingProfile) {
  return <GordonLoader />;
} else {
// render transcript
}
```